common/concurrent.go

```diff
@@ -0,0 +1,35 @@
+package common
+
+import "sync"
+
+// Concurrent Safe String keyed map
+type StringKeyedMap[T any] struct {
+	items map[string]T
+	lock  *sync.RWMutex
+}
+
+func NewStringKeyedMap[T any]() *StringKeyedMap[T] {
```

Comment:
There's no need to return a pointer, right?

Reply:
Makes sense. I'll make it return by value.